lib/client/api.go

```diff
@@ -1933,6 +1933,15 @@ func (tc *TeleportClient) runShell(nodeClient *NodeClient, sessToJoin *session.S
 		return trace.Wrap(err)
 	}
 	if err = nodeSession.runShell(tc.OnShellCreated); err != nil {
+		switch e := trace.Unwrap(err).(type) {
+		case *ssh.ExitError:
+			tc.ExitStatus = e.ExitStatus()
+		case *ssh.ExitMissingError:
+			tc.ExitStatus = 1
```

Comment:
Not sure if this is semantically correct. Must `ExitMissingError` imply a failed/non-zero `ExitStatus`?

Reply:
I copied the behavior of non interactive sessions: https://github.com/gravitational/teleport/blob/master/lib/client/api.go#L1918